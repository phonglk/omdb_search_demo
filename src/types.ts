import { ReactNode } from 'react';

export type ComponentProps = {
  children: ReactNode;
};

export type State = {
  searchTerm: string;
  isSearching: boolean;
  searchResult: Movie[];
};

export type Movie = {
  title: string;
  poster: string;
  year: number;
  id: string;
};

export type APIResponse = {
  Error?: string;
  Search?: Array<{
    Poster: string;
    Title: string;
    Year: string;
    imdbID: string;
  }>;
};
