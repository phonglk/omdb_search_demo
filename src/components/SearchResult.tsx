import React from 'react';
import styled from 'styled-components';
import { Movie } from '../types';

type Props = {
  movies: Movie[];
};

const Wrapper = styled.div`
  display: grid;
  grid-template-columns: 1fr 1fr 1fr 1fr 1fr;
  column-gap: 20px;
  row-gap: 20px;
  margin: 20px 0;
`;

const Year = styled.div`
  position: absolute;
  right: 0;
  top: 5px;
  background: rgb(225 29 72);
  color: white;
  font-size: 14px;
  border-top-left-radius: 10px;
  border-bottom-left-radius: 10px;
  padding: 3px 5px 3px 10px;
  font-weight: bold;
`;

const Title = styled.div`
  position: absolute;
  left: 0;
  bottom: -100%;
  background: rgb(37 99 235);
  color: white;
  font-size: 14px;
  padding: 5px 2px;
  text-align: center;
  font-weight: bold;
  width: 100%;
  box-sizing: border-box;
  transition: all 0.2s;
  opacity: 0;
`;

const MovieWrapper = styled.div`
  display: block;
  position: relative;
  width: 100%;
  height: 230px;
  background-position: center center;
  background-size: cover;
  overflow: hidden;
  border-radius: 10px;
  filter: drop-shadow(0 10px 10px rgba(0, 0, 0, 0.15));
  transition: all 0.2s;
  border: 1px solid rgba(0, 0, 0, 0.3);

  &:hover {
    transform: scale(1.1);
    filter: drop-shadow(0 10px 20px rgba(0, 0, 0, 0.25));
    border: 1px solid rgba(0, 0, 0, 0.8);
    ${Title} {
      bottom: 0;
      opacity: 1;
    }
  }
`;

export default function SearchResult({ movies }: Props) {
  return (
    <Wrapper>
      {movies.map((movie) => (
        <MovieWrapper
          key={movie.id}
          style={{ backgroundImage: `url(${movie.poster})` }}
        >
          <Year>{movie.year}</Year>
          <Title>{movie.title}</Title>
        </MovieWrapper>
      ))}
    </Wrapper>
  );
}
