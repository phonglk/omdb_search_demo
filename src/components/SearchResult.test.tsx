import React from 'react';
import { render } from '@testing-library/react';
import SearchResult from './SearchResult';
import { Movie } from '../types';

test('Search Result render correctly', () => {
  const movies: Movie[] = [
    {
      title: 'Movie 1',
      year: 2019,
      poster: 'poster1',
      id: 'tt00001',
    },
    {
      title: 'Movie 2',
      year: 2019,
      poster: 'poster2',
      id: 'tt00002',
    },
    {
      title: 'Movie 3',
      year: 2019,
      poster: 'poster3',
      id: 'tt00003',
    },
    {
      title: 'Movie 4',
      year: 2019,
      poster: 'poster4',
      id: 'tt00004',
    },
  ];

  const { container } = render(<SearchResult movies={movies} />);
  expect(container).toMatchSnapshot();
});
