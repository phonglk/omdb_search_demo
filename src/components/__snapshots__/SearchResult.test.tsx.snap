// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Result render correctly 1`] = `
<div>
  <div
    class="sc-bczRLJ bjGgPt"
  >
    <div
      class="sc-hKMtZM iyMRBL"
      style="background-image: url(poster1);"
    >
      <div
        class="sc-gsnTZi eoCyxL"
      >
        2019
      </div>
      <div
        class="sc-dkzDqf dNNbLI"
      >
        Movie 1
      </div>
    </div>
    <div
      class="sc-hKMtZM iyMRBL"
      style="background-image: url(poster2);"
    >
      <div
        class="sc-gsnTZi eoCyxL"
      >
        2019
      </div>
      <div
        class="sc-dkzDqf dNNbLI"
      >
        Movie 2
      </div>
    </div>
    <div
      class="sc-hKMtZM iyMRBL"
      style="background-image: url(poster3);"
    >
      <div
        class="sc-gsnTZi eoCyxL"
      >
        2019
      </div>
      <div
        class="sc-dkzDqf dNNbLI"
      >
        Movie 3
      </div>
    </div>
    <div
      class="sc-hKMtZM iyMRBL"
      style="background-image: url(poster4);"
    >
      <div
        class="sc-gsnTZi eoCyxL"
      >
        2019
      </div>
      <div
        class="sc-dkzDqf dNNbLI"
      >
        Movie 4
      </div>
    </div>
  </div>
</div>
`;
